### PR TITLE
Added /tags/ endpoint

### PR DESCRIPTION
**Pull request by bholagabbar**
Me and @TheHollidayInn had a long conversation on how search could be improved in the SEdaily apps. One possible way was to display a whole bunch of all the tags to the user and let them choose the tags they find interesting and return the posts that match the intersection of all these tags.

The current api does support filtering of tags https://github.com/SoftwareEngineeringDaily/software-engineering-daily-api/blob/master/server/models/post.model.js#L90

However, there was no endpoint in the API which showed the user/mobile developers all the tags in the system hence they were unable to list them onto the app.

I have added the `/api/tags` and `/api/tags/:tagId` endpoints. I've tested them on my system and am pasting screenshots from Postman as well. Feel free to pull in these changes and test them on your system before merging. 

`/api/tags`

`/api/tags/:tagId`

cc @jasonify @crablar

### CODE REVIEW

File: server/tests/tag.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+xdescribe('## Tag APIs', () => {
+
+  describe('# GET /api/users/:userId', () => {
+    it('should get user details', (done) => {
+      request(app)
+        .get('/api/tags/')
+        .expect(httpStatus.OK)
+        .then((res) => {
+          done();
```
Comment (TheHollidayInn) [inline server/tests/tag.test.js]: This needs to check that tags are returned.

File: server/tests/tag.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  mongoose.models = {};
+  mongoose.modelSchemas = {};
+  mongoose.connection.close();
+  done();
+});
+
+xdescribe('## Tag APIs', () => {
+
+  describe('# GET /api/users/:userId', () => {
+    it('should get user details', (done) => {
```
Comment (TheHollidayInn) [inline server/tests/tag.test.js]: Can you update the text to speak about tags and not user?

File: server/tests/tag.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      request(app)
+        .get('/api/tags/')
+        .expect(httpStatus.OK)
+        .then((res) => {
+          done();
+        })
+        .catch(done);
+    });
+
+    it('should report error with message - Not found, when tag does not exists', (done) => {
```
Comment (TheHollidayInn) [inline server/tests/tag.test.js]: You need to write a task for the tag detail route as well

File: server/models/tag.model.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+   * @returns {Promise<Tag, APIError>}
+   */
+  get(id) {
+    return this.findOne({ id: id })
+      .exec()
+      .then((tag) => {
+        if (tag) {
+          return tag;
+        }
+        const err = new APIError('No such user exists!', httpStatus.NOT_FOUND);
```
Comment (TheHollidayInn) [inline server/models/tag.model.js]: This should say tag not user

File: server/tests/tag.test.js (hunk 3 above)
Comment (bholagabbar) [reply server/tests/tag.test.js]: @TheHollidayInn You mean adding a test for `api/tags/` right? Have done that

Comment (TheHollidayInn) [reply server/tests/tag.test.js]: Nah, tag detail is `api/tags/:tagid`. Looks like you added that though.

File: server/tests/tag.test.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+ */
+after((done) => {
+  // required because https://github.com/Automattic/mongoose/issues/1251#issuecomment-65793092
+  mongoose.models = {};
+  mongoose.modelSchemas = {};
+  mongoose.connection.close();
+  done();
+});
+
+xdescribe('## Tag APIs', () => {
```
Comment (TheHollidayInn) [inline server/tests/tag.test.js]: Can you remove this from being pending?

File: server/tests/tag.test.js (hunk 3 above)
Comment (bholagabbar) [reply server/tests/tag.test.js]: @TheHollidayInn Did that, removed the wait and fixed the tests. All tests run successfully :)

**Comment by TheHollidayInn** [comment]
Great job!

**Comment by crablar** [comment]
@bholagabbar Nice work!

**Pull request merged**